packages/nimble-components/src/table/models/table-validator.spec.ts

```diff
@@ -1,234 +1,353 @@
-    it('ID field name can be an empty string', () => {
-        const data = [
-            // eslint-disable-next-line @typescript-eslint/naming-convention
-            { stringField: 'value-1', numberField: 10, '': 'empty-1' },
-            // eslint-disable-next-line @typescript-eslint/naming-convention
-            { stringField: 'value-2', numberField: 11, '': 'empty-2' }
+    describe('column ID validation', () => {
+        const columnConfigurations: {
+            columnIds: (string | undefined)[],
+            invalidKeys: (keyof TableValidity)[],
+            testDescription: string
+        }[] = [
+            {
+                columnIds: [undefined, ''],
+                invalidKeys: [],
+                testDescription: 'does not require column IDs'
+            },
+            {
+                columnIds: ['my-id', undefined, undefined],
+                invalidKeys: ['missingColumnId'],
+                testDescription:
+                    'requires column IDs for all columns if a column ID is defined for at least one'
+            },
+            {
+                columnIds: ['my-id-1', 'my-id-2', 'my-id-3'],
+                invalidKeys: [],
+                testDescription: 'unique defined IDs are valid'
+            },
+            {
+                columnIds: ['my-id-1', 'my-id-2', 'my-id-2'],
+                invalidKeys: ['duplicateColumnId'],
+                testDescription: 'duplicate column IDs is invalid'
+            },
+            {
+                columnIds: ['my-id-1', 'my-id-2', 'my-id-2', undefined],
+                invalidKeys: ['missingColumnId', 'duplicateColumnId'],
+                testDescription: 'reports multiple column ID validation errors'
+            },
+            {
+                columnIds: ['my-id-1', ''],
+                invalidKeys: ['missingColumnId'],
+                testDescription:
+                    'does not allow empty string as a defined column ID'
+            }
         ];
-        const validator = new TableValidator();
-
-        const isValid = validator.validateRecordIds(data, '');
-        expect(isValid).toBeTrue();
-        expect(validator.isValid()).toBeTrue();
-        const validity = validator.getValidity();
-        expect(validity.duplicateRecordId).toBeFalse();
-        expect(validity.invalidRecordId).toBeFalse();
-        expect(validity.missingRecordId).toBeFalse();
+
+        for (const columnConfiguration of columnConfigurations) {
+            // eslint-disable-next-line @typescript-eslint/no-loop-func
+            it(columnConfiguration.testDescription, () => {
+                const isValid = validator.validateColumnIds(
+                    columnConfiguration.columnIds
+                );
+                expect(isValid).toBe(
+                    columnConfiguration.invalidKeys.length === 0
+                );
+                expect(validator.isValid()).toBe(
+                    columnConfiguration.invalidKeys.length === 0
+                );
```

Comment:
Can we avoid the conditional expects like above? Maybe in the test config we can include the expected boolean state, i.e. columnConfiguration.isValid? Just trying to stick to declarative expects